dsl/pace/dsl/caches/cache_location.py

```diff
@@ -0,0 +1,46 @@
+from pace.dsl.caches.codepath import FV3CodePath
+from pace.util import CubedSpherePartitioner
+
+
+def identify_code_path(
```

Comment:
Something about this feels awkward to me, but I'm not sure making this live inside of a more fully-featured FV3CodePath class makes sense?

Reply:
We need a proper build class/system that uproots most of all of this and consolidate it to one place. It should include distributed compiling for non-orchestrated backend, better hash/cache and much more.

I went for a functional paradigm in the meantime, since I fully expect this to be reworked

Reply:
Yeah I don't have a super compelling alternative and it's not really blocking, just felt awkward when I read it